### PR TITLE
Adding terraform provider version to user-agent header

### PR DESCRIPTION
**Pull request by matpimenta**

### CODE REVIEW

File: GNUmakefile [hunk 1]
```diff
@@ -2,19 +2,20 @@ TEST?=$$(go list ./... |grep -v 'vendor')
 GOFMT_FILES?=$$(find . -name '*.go' |grep -v vendor)
 WEBSITE_REPO=github.com/hashicorp/terraform-website
 PKG_NAME=fastly
+VERSION=$(shell git describe --tags --always)
 
 default: build
 
 build: fmtcheck
-	go install
+	go install -ldflags="-X github.com/terraform-providers/terraform-provider-fastly/version.ProviderVersion=$(VERSION)"
```
Comment (phamann) [inline GNUmakefile]: I suggest moving the repo string `github.com/terraform-providers/terraform-provider-fastly/` to a variable at the top of the file to use in both of these lines.

**Review by phamann — APPROVED** [review]
:+1: LGTM other than minor nit.

File: GNUmakefile (hunk 1 above)
Comment (matpimenta) [reply GNUmakefile]: good point, let me fix that

**Review by trentrosenbaum — APPROVED** [review]

File: fastly/config.go [hunk 2]
```diff
@@ -24,7 +27,9 @@ func (c *Config) Client() (interface{}, error) {
 		return nil, fmt.Errorf("[Err] No API key for Fastly")
 	}
 
-	gofastly.UserAgent = terraform.UserAgentString()
+	composedUserAgent := fmt.Sprintf("%s %s/%s", terraform.UserAgentString(), TerraformProviderProductUserAgent, version.ProviderVersion)
```
Comment (philippschulte) [inline fastly/config.go]: `terraform.UserAgentString()` is deprecated!

**Review by philippschulte — CHANGES_REQUESTED** [review]
I have looked at the big public cloud providers how they set the `User-Agent` header and noticed that `terraform.UserAgentString()` is deprecated! I would prefer the same pattern used by the big providers:

```
tfUserAgent := httpclient.UserAgentString()
pv := version.ProviderVersion
providerUserAgent := fmt.Sprintf("%s terraform-provider-fastly/%s", tfUserAgent, pv)
gofastly.UserAgent = providerUserAgent
```

**Comment by matpimenta** [comment]
I've removed the deprecation and changed user agent to `terraform-provider-fastly`.

**Review by philippschulte — APPROVED** [review]

**Pull request merged**